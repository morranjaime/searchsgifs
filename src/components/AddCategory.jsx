
import { useState } from 'react';
import Proptypes from 'prop-types';

export const AddCategory = (props) => {

    const { setCategories, currentCategories } = props;// para pasar el estado

    // const { onAddNewCategory } = props;

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length >= 1 && !currentCategories.map(category => category.toUpperCase()).includes(inputValue.toUpperCase())) {
            setCategories([inputValue, ...currentCategories]);
            //setCategories(categories =>[inputValue, ...categories]);
            // onAddNewCategory(inputValue)
            setInputValue('');
        }

    }

    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label='form'>
            <input type='text' placeholder='Buscar gif' value={inputValue} onChange={(event) => onInputChange(event)} />
        </form>
    )
}


AddCategory.propTypes =
{
    setCategories: Proptypes.func.isRequired,
    currentCategories: Proptypes.array.isRequired
}