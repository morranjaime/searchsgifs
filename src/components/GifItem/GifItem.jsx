
export const GifItem = (props) => {

    const { title, url } = props;
    return (
        <div className='card'>
            <img src={url} title={title} />
            <p>{title}</p>
        </div>
    )
}
