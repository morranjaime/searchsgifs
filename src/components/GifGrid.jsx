import React from 'react'
import Proptypes from 'prop-types';
import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';
import { useFecthGifs } from '../hooks/useFecthGifs';

export const GrifGrid = (props) => {

    const { category } = props;

    const { images, isLoading } = useFecthGifs(category);

    return (
        <>
            <h2> {category}</h2>
            {isLoading && <h2>Cargando....</h2>}
            <div className='card-grid'>
                {images.map((image) => <GifItem key={image.id} {...image} />)}
            </div>
        </>

    )
}

GrifGrid.propTypes =
{
    category: Proptypes.string.isRequired
}
