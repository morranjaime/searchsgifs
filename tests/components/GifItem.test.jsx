import { render, screen, fireEvent } from '@testing-library/react'
import { GifItem } from "../../src/components/GifItem"

describe('Prueba en GifItem', () => {

    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg'

    test('Realizacion de snapshp', () => {
        const { container } = render(
            <GifItem title={title} url={url} />
        );

        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
        render(
            <GifItem title={title} url={url} />
        );

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => {

        render(
            <GifItem title={title} url={url} />
        );

        expect(screen.getByText(title)).toBeTruthy();

    });
})