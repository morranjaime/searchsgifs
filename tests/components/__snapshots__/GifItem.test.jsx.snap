// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifItem Realizacion de snapshp 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
